### PR TITLE
Rule for raw socket creation

### LINKED ISSUE #179: Raw socket creation
## Summary

Creation of raw sockets.

## Examples

`WSASocketA(2, 3, 0, 0, 0, 1)`

## Features

socket functions
constants, see https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket, https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasocketa

### PR DESCRIPTION
**Pull request by bkojusner**
This rule passes the thorough linting test and closes #179

### CODE REVIEW

File: communication/socket/create-raw-socket-in-windows.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: create raw socket in Windows
```
Comment (mr-tz) [inline communication/socket/create-raw-socket-in-windows.yml:4]: similar socket rules are OS independent, I'd suggest to do the same here

Comment (mr-tz) [inline communication/socket/create-raw-socket-in-windows.yml:4]:
```suggestion
    name: create raw socket
```

File: communication/socket/create-raw-socket-in-windows.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - https://learn.microsoft.com/en-us/windows/win32/winsock/tcp-ip-raw-sockets-2
+      - https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket
+      - https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasocketa
+    examples:
+      - 10EBCF8C20403457A08762200015B151:0x140001000
+  features:
+    - and:
+      - or:
+        - api: socket
+        - api: WSASocketA
```
Comment (mr-tz) [inline communication/socket/create-raw-socket-in-windows.yml:19]:
```suggestion
        - api: ws2_32.WSASocketA
```

File: communication/socket/create-raw-socket-in-windows.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - or:
+        - operand[1].number: 2 = AF_INET
+        - operand[1].number: 23 = AF_INET6
+      - operand[1].number: 3 = SOCK_RAW
```
Comment (mr-tz) [inline communication/socket/create-raw-socket-in-windows.yml:23]:
```suggestion
        - number: 2 = AF_INET
        - number: 23 = AF_INET6
      - number: 3 = SOCK_RAW
```

Comment (mr-tz) [reply communication/socket/create-raw-socket-in-windows.yml:23]: otherwise, `push 2` would not fire

**Review by mr-tz — CHANGES_REQUESTED** [review]

File: communication/socket/create-raw-socket-in-windows.yml (hunk 3 above)
Comment (williballenthin) [reply communication/socket/create-raw-socket-in-windows.yml:23]: we need call scope!

Comment (mr-tz) [reply communication/socket/create-raw-socket-in-windows.yml:23]: coming soon ™️ ?

File: communication/socket/create-raw-socket-in-windows.yml (hunk 1 above)
Comment (bkojusner) [reply communication/socket/create-raw-socket-in-windows.yml]: Implemented change in `name`.

Comment (bkojusner) [reply communication/socket/create-raw-socket-in-windows.yml]: Thanks for the suggestion! Renamed file and implementing the additional feedback.

File: communication/socket/create-raw-socket-in-windows.yml (hunk 3 above)
Comment (bkojusner) [reply communication/socket/create-raw-socket-in-windows.yml]: Would definitely be a great addition for items like these. Thanks for all the feedback, rule is updated accordingly.

**Review by mr-tz — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**